packages/@expo/cli/src/start/platforms/ExpoGoInstaller.ts

```diff
@@ -74,6 +76,16 @@ export class ExpoGoInstaller<IDevice> {
   async ensureAsync(deviceManager: DeviceManager<IDevice>): Promise<boolean> {
     let shouldInstall = !(await deviceManager.isAppInstalledAsync(this.appId));
 
+    if (APISettings.isOffline && !shouldInstall) {
+      Log.warn(`Skipping Expo Go version validation in offline mode`);
+      return false;
+    } else if (APISettings.isOffline && shouldInstall) {
+      throw new CommandError(
+        'NO_EXPO_GO',
+        `Expo Go is not installed on device "${deviceManager.name}", while running in offline mode. Manually install Expo Go or run without --offline flag.`
+      );
```

Comment:
I think this is the best context we can give them. Pointing to an `expo.fyi` seems pointless in offline mode's case 🤷